crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs

```diff
@@ -124,13 +125,20 @@ impl KitsuneP2pActor {
             }
         };
 
+        // wrap in restart logic
+        let f = tx2_restart_adapter(f);
+
         // convert to frontend
         let f = tx2_pool_promote(f, config.tuning_params.clone());
 
         // wrap in proxy
         let f = if !is_mock {
             let mut conf = kitsune_p2p_proxy::tx2::ProxyConfig::default();
             conf.tuning_params = Some(config.tuning_params.clone());
+            if let Some(use_proxy) = &tx2_conf.use_proxy {
+                let proxy_url = ProxyUrl::from(use_proxy.as_str());
+                conf.client_of_remote_proxy = Some(proxy_url);
+            }
```

Comment:
This allows us to delete all the code below in this file. It makes more sense to manage the proxy connection resiliency inside the proxy wrapper logic, we can do it much more efficiently there.